WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -43,26 +54,55 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
-    /** Instantiate the interface and set the context  */
-    class WebAppInterface(private val mContext: Context) {
-        /** Send a message from the web page  */
-        @JavascriptInterface
-        fun sendMessage(message: String) {
-            val sendIntent: Intent = Intent().apply {
-                action = Intent.ACTION_SEND
-                putExtra(Intent.EXTRA_TEXT, message)
-                type = "text/plain"
-            }
-            val shareIntent = Intent.createChooser(sendIntent, null)
-            startActivity(mContext, shareIntent, null)
+    private fun createJsObject(
+        webview: WebView,
+        mContext: Context,
+        jsObjName: String,
+        rules: Set<String>,
+        sendAndroidMessage: (message: String, mContext: Context) -> Unit
+    ) {
+        if (WebViewFeature.isFeatureSupported(WebViewFeature.WEB_MESSAGE_LISTENER)) {
+            WebViewCompat.addWebMessageListener(
+                webview, jsObjName, rules,
+                object : WebViewCompat.WebMessageListener {
+                    override fun onPostMessage(
+                        webview: WebView,
+                        message: WebMessageCompat,
+                        sourceOrigin: Uri,
+                        isMainFrame: Boolean,
+                        replyProxy: JavaScriptReplyProxy
+                    ) {
+                        sendAndroidMessage(message.data, mContext)
+                    }
+                })
+        } else {
+            webview.addJavascriptInterface(object {
+                @JavascriptInterface
+                fun postMessage(message: String?) {
+                    sendAndroidMessage(message, mContext)
+                }
+            }, jsObjName)
         }
+    }
 
+    // Invokes native android sharing
+    val sendAndroidMessage = { message: String?, mContext: Context ->
```

Comment:
nit: `mContext` should just be `context`

Reply:
ok